### ISSUE TITLE
Custom Asset for custom Modernizr build?

### ISSUE THREAD

**Issue by hirasso** [parcel-bundler/parcel#1987]
# ❔ Question

Does any of you guys have an idea on how to create a proper parcel plugin for a custom Modernizr build in parcel? Just a general direction, like "do this, and then that, ..." would be already great as a starting point.

## 🔦 Context

I have been happily coding my parcel plugin for generating a custom Mordernizr build [over here](https://github.com/hirasso/parcel-plugin-modernizr), just to realize today that I have been doing it wrong big time 😅

At the moment, the plugin simply hooks into the `bundled`-event and creates the custom modernizr build inside the `/dist`-folder. If using `.js`-files as entryPoints for parcel and loading the bundled assets from another environment like in my case WordPress, this works just fine. 

But if I would use an `index.html` inside the `/src`-folder as entryFile and also use this file as my main app entry in `/dist`, this approach wouldn't work. 

I'm pretty sure there is a ✨golden way✨ to solve this and think I could learn a lot more about parcel's way of doing things on the way.

**Comment by hirasso** [comment]
Moved main code from `generate()` to `parse()`

**Comment by hirasso** [comment]
Hmm, you are right. This could be fixed by extending the `JSAsset`, right?

**Comment by DeMoorJasper** [comment]
We strongly advice against extending internal assets

**Comment by hirasso** [comment]
Hrrm, ok. Any advise then on how I should do this? Create a new file type "custom.modernizr"? I need some way to embed the script in the final rendered index.html...

**Comment by DeMoorJasper** [comment]
I guess that's the way to go, not sure how else to do this besides pre-generating it before running parcel

**Comment by hirasso** [comment]
OK, I first tried to pre-generate the modernizr.js in the `./src` folder on `buildStart`, but that seems to be too late for parcel to find the file, right? A really early hook, with just the options being defined, would be great. If this is not possible, I might really go with the `custom.modernizr` option.

**Comment by DeMoorJasper** [comment]
I meant as a command before parcel, not actually as a plugin.

But yeah there is no way to add stuff to the mainBundle reliably using buildStart

**Comment by hirasso** [comment]
> I meant as a command before parcel, not actually as a plugin.

Right, I was doing that before I got completely lost in the parcel API. Just this thing we sometimes like to do, waste a lot of time on writing a ton of code just to save one line of code somewhere else 😜

Anyways, I already learnt a lot! ...and since I started it already, I might go the distance and do this with a `custom.modernizr` Asset file. Thank you so much for your time and patience.
